packages/orm/src/EntityManager.ts

```diff
@@ -459,6 +459,32 @@ export class EntityManager<C extends HasKnex = HasKnex> {
     return entities as T[];
   }
 
+  /**
+   * Returns instances of `type` for the given `ids`, resolving to an existing instance if in our Unit of Work. Ignores
+   * IDs that are not found.
+   */
+  public async loadAllIfExists<T extends Entity>(type: EntityConstructor<T>, ids: string[]): Promise<T[]>;
+  public async loadAllIfExists<
+    T extends Entity,
+    H extends LoadHint<T> & ({ [k: string]: N | H | [] } | N | N[]),
+    N extends Narrowable
+  >(type: EntityConstructor<T>, ids: string[], populate: H): Promise<Loaded<T, H>[]>;
+  async loadAllIfExists<T extends Entity>(type: EntityConstructor<T>, _ids: string[], hint?: any): Promise<T[]> {
+    const meta = getMetadata(type);
+    const ids = _ids.map((id) => tagIfNeeded(meta, id));
+    const entities = (
+      await Promise.all(
+        ids.map((id) => {
+          return this.findExistingInstance(id) || this.loaderForEntity(meta).load(id);
+        }),
```

Comment:
curious: I haven't looked around here, but I was intuitively thinking this would translate to some sort of `select * from blah where id in [ids]` (and perhaps it is eventually?).

What's the reasoning for doing it one at a time? I guess we are checking the cache first, then loading what we need. Would there be a benefit to loading things that aren't cached like suggested compared to separate queries? Perhaps the query manager is performing that magic under the hoods?

Reply:
`loaderForEntity` uses `DataLoader` under the hood which handles the batching (I think)

Reply:
Yep